### PR TITLE
Created Gmail cheat sheet

### PR DESCRIPTION
**Pull request by sagarhani**
# Goodie Pull Request Template

Using this template will help us better understand your Instant Answer and assist you when necessary.  Simply copy and paste the markdown below into the description of your GitHub pull request and complete as appropriate.

**What does your Instant Answer do?**
Shows the shortcut keys to work in Gmail, google mailing client.

**What problem does your Instant Answer solve (Why is it better than organic links)?**
It is easier to use gmail with these shortcut keys.

**What is the data source for your Instant Answer? (Provide a link if possible)**
Got the data from make us of website - http://www.makeuseof.com/downloads/gmail-cheat-sheet.pdf

**Why did you choose this data source?**
It's very accurate and precise.

**What are some example queries that trigger this Instant Answer?**
gmail cheat sheet
gmail cheatsheet

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**
students, working professionals, all internet users.

**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**
No.

**Which existing Instant Answers will this one supercede/overlap with?**
None.

**Are you having any problems? Do you need our help with anything?**
No.

**Where did you hear about DuckDuckHack? (For first time contributors)**

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**
gmail-cheat-sheet

## Checklist
Please place an 'X' where appropriate.

```
- [] Added metadata and attribution information (https://duck.co/duckduckhack/metadata)
- [] Wrote test file and added to t/ directory (https://duck.co/duckduckhack/goodie_tests)
- [] Verified that instant answer adheres to design guidelines (https://duck.co/duckduckhack/design_styleguide)
- [X] Verified that instant answer adheres to code styleguide (https://duck.co/duckduckhack/code_styleguide)
- [X] Tested cross-browser compatibility

    Please let us know which browsers/devices you've tested on:
    -Linux (ubuntu 14.04)
        - [X] Firefox
```

IA Page: https://duck.co/ia/view/gmail_cheat_sheet

### CODE REVIEW

**Comment by MariagraziaAlastra** [comment]
Hi @sagarhani, thanks for your contribution, I'm checking it out :)

File: lib/DDG/Goodie/WhereAmI.pm [hunk 1]
```diff
@@ -1,35 +0,0 @@
...
-
-    return $answer,
-      structured_answer => {
-        input     => [],
-        operation => 'Apparent current location',
-        result    => $answer
-      };
-};
-
-1;
```
Comment (MariagraziaAlastra) [inline lib/DDG/Goodie/WhereAmI.pm]: @sagarhani looks like WhereAmI.pm is accidentally deleted in one of your commits. Could you please restore it? Thanks!

File: lib/DDG/Goodie/RouterPasswords.pm [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package DDG::Goodie::RouterPasswords;
+    package DDG::Goodie::RouterPasswords;
```
Comment (MariagraziaAlastra) [inline lib/DDG/Goodie/RouterPasswords.pm]: @sagarhani I guess you didn't really want to commit this change :)

**Comment by sagarhani** [comment]
@MariagraziaAlastra Oops! Yes I didn't wanted to commit this change. I didn't know how this happend. Is it necessary to fix that ? Instead Merge only gmail.json file. Ignore that file.

**Comment by MariagraziaAlastra** [comment]
@sagarhani When a PR is merged all the changes in the commits are included, so yeah, it needs to be fixed.
Don't worry, it's quite common to accidentally include changes in a commit, we've all been there (more than just once, too) :)

**Comment by sagarhani** [comment]
@MariagraziaAlastra I guess I have fixed it now. Is it proper now ?

**Comment by MariagraziaAlastra** [comment]
Yup, thanks @sagarhani!
I'll ask for the design team signoff before merging :)

**Comment by sagarhani** [comment]
Cool! :sunglasses: 
Some more cheat sheets will be coming up. :+1:

File: share/goodie/cheat_sheets/gmail.json [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        "Shortcuts",
+        "Navigate"
+    ],
+    "sections": {
+        "Shortcuts": [{
+            "key": "c",
+            "val": "Compose a new message"
+            
+        }, {
+            "key": "[Shift] + [c]",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/gmail.json]: We've just published updated docs and we're now asking that you remove the `+` from Cheat Sheets, so this should be: `[Shift] [C]`

Comment (moollaza) [inline share/goodie/cheat_sheets/gmail.json]: Sorry about the confusoin -- we'll need to update the live examples@

File: share/goodie/cheat_sheets/gmail.json [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+            "key": "[Shift] + [f]",
+            "val": "Forward a message in a new window"
+            
+        }, {
+            "key": "Esc",
+            "val": "Escape from input field"
+            
+            }],
+        "Navigate": [{
+            "key": "[Tab] then [Enter]",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/gmail.json]: Similarly, successive key presses should be separated wih a comma, `,`:

```
[Tab], [Enter]
```

**Comment by sagarhani** [comment]
I'll fix this now! :+1:

**Comment by sagarhani** [comment]
@moollaza can you check now ?

**Comment by sagarhani** [comment]
@moollaza  or @MariagraziaAlastra  does this require any more changes ?

**Comment by moollaza** [comment]
@sagarhani whoops! It looks like you've remove the branch that was used to create this PR? I can't pull down and test the code!

File: share/goodie/cheat_sheets/gmail.json [hunk 5]
```diff
@@ -0,0 +1,117 @@
+{
+    "id": "gmail_cheat_sheet",
+    "name": "Gmail",
+    "description": "Mailing Client",
+    "metadata": {
+        "sourceName": "GmailCheatSheet",
+        "sourceUrl": "https://duckduckgo.com"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/gmail.json]: Let's put the actual source of this info!

**Comment by moollaza** [comment]
@sagarhani aside from the source, this is looking pretty good!

**Comment by moollaza** [comment]
@sagarhani congrats! It's live: https://duckduckgo.com/?q=gmail+cheat+sheet&ia=cheatsheet

Sorry for the delay, it's actually been live for a while now!

Thanks again for taking the time to contribute! We really appreciate it. Feel free to stick around and comment/help on other PR's, tackle open issues or even submit more Instant Answers!

**Pull request merged**